kroxylicious-kms-provider-fortanix-dsm/src/test/java/io/kroxylicious/kms/provider/fortanix/dsm/FortanixDsmKmsEdekSerdeTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.fortanix.dsm;
+
+import java.nio.ByteBuffer;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.kroxylicious.kms.service.Serde;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class FortanixDsmKmsEdekSerdeTest {
```

Comment:
maybe we should add a v0 decode test that takes bytes as input, something that would fail if we changed v0 decoding.